setup.sh

```diff
@@ -60,6 +82,49 @@ function check_dependencies()
   fi
 }
 
+# TODO
+# Originally, kw get installed in the ~/.config and add the kw binary in the
+# path. We changed it; however, we keep this function for supporting the
+# migration from the old version to the new one.
+function remove_kw_from_PATH_variable()
+{
+  local new_path=""
+  local needs_update=0
+
+  IFS=':' read -ra ALL_PATHS <<< "$PATH"
+  for path in "${ALL_PATHS[@]}"; do
+    if [[ "$path" =~ "/kw" ]]; then
+      needs_update=1
+      continue
+    fi
+    # The first interaction introduce one extra ':'
+    new_path="$new_path:$path"
+  done
+
+  if [[ "$needs_update" != 0 ]]; then
+    # Drop ':' introduced in the above loop
+    PATH="${new_path:1}"
+    export PATH
+  fi
+}
+
+function update_path()
+{
+  local new_path=""
+
+  IFS=':' read -ra ALL_PATHS <<< "$PATH"
+  for path in "${ALL_PATHS[@]}"; do
+    [[ "$path" -ef "$binpath" ]] && return
+  done
+
+  echo "PATH=$HOME/.local/bin:\$PATH" >> "$HOME/.bashrc"
```

Comment:
Would be 'echo "PATH=\$PATH:$kwbinpath" ' ?

Reply:
Even after modifying it, as mentioned above, kw command is not found. It is still looking for this kind of pattern /home/$USER/.config/kw/ ... not sure if there is some trash from the current version or what, but I got stuck on it.